Tryouts/Core/BasicModels/Modules/ProcessesViewModel.cs

```diff
@@ -0,0 +1,208 @@
+/// See the License for the specific language governing permissions and limitations under the License.
+/// 
+/// ********************************************************************************************************
+
+using MorganStanley.ComposeUI.Tryouts.Core.Abstractions.Modules;
+using System.Collections.ObjectModel;
+using System.Reactive.Linq;
+using NP.Concepts.Behaviors;
+using NP.Utilities;
+
+namespace MorganStanley.ComposeUI.Tryouts.Core.BasicModels.Modules
+{
+    public class ProcessesViewModel
+    {
+        public ModuleViewModel[] Modules { get; }
+
+        IModuleLoader _moduleLoader;
+
+        public IEnumerable<SingleProcessViewModel> Processes { get; } =
+            new ObservableCollection<SingleProcessViewModel>();
+
+
+        public IEnumerable<SingleProcessViewModel> ProcessesWithWindows { get; } =
+            new ObservableCollection<SingleProcessViewModel>();
+
+        private IDisposable _modulesBehavior;
+        private IDisposable _processesBehavior;
+        IModuleLoaderFactory _moduleLoaderFactory;
+        public ProcessesViewModel
+        (
+            IModuleLoaderFactory moduleLoaderFactory,
+            ModuleViewModel[] modules)
+        {
+            _moduleLoaderFactory = moduleLoaderFactory;
+
+            Modules = modules;
+
+            _modulesBehavior = Modules.AddBehavior(OnModuleAdded, OnModuleRemoved);
+
+            _processesBehavior = Processes.AddBehavior(OnProcessesAdded, OnProcessesRemoved);
+
+            ModuleCatalogue moduleCatalogue =
+                new ModuleCatalogue(Modules.ToDictionary(m => m.Name, m => (ModuleManifest)m));
+
+            _moduleLoader = _moduleLoaderFactory.Create(moduleCatalogue);
+
+            _moduleLoader.LifecycleEvents.Subscribe(OnLifecycleEventArrived);
+        }
+
+        internal void ResetProcesses()
+        {
+            (Processes as IList<SingleProcessViewModel>).RemoveAllOneByOne();
+        }
+
+        private void OnLifecycleEventArrived(LifecycleEvent lifecycleEvent)
+        {
+            Guid instanceId = lifecycleEvent.ProcessInfo.instanceId;
+
+            var process = Processes.FirstOrDefault(p => p.InstanceId == instanceId);
+
+            process?.ReactToMessage(lifecycleEvent);
+        }
+
+        private void OnModuleAdded(ModuleViewModel module)
+        {
+            module.LaunchEvent += OnLaunchModule;
+        }
+
+        private void OnModuleRemoved(ModuleViewModel module)
+        {
+            module.LaunchEvent -= OnLaunchModule;
+        }
+
+        private void OnProcessesAdded(SingleProcessViewModel process)
+        {
+            process.StopEvent += OnStopProcess;
+            process.StartedEvent += Process_StartedEvent;
+            process.StoppedEvent += Process_StoppedEvent;
+        }
+
+        private void Process_StoppedEvent(SingleProcessViewModel process)
+        {
+            ((IList<SingleProcessViewModel>)ProcessesWithWindows).Remove(process);
```

Comment:
added private _processes and _prossessesWithWindows of type `Observable<SingleProcessViewModel>` so that I could remove the castings. 
The properties Processes and ProcessesWithWindows are still of `IEnumerable<SingleProcessViewModel>` so that no client of the class would be able to modify them.